.gitlab/kernel_matrix_testing/common.yml

```diff
@@ -169,6 +169,17 @@
     - scp -o StrictHostKeyChecking=no -i $AWS_EC2_SSH_KEY_FILE "ubuntu@$INSTANCE_IP:/tmp/ddvm-xml-*" $CI_PROJECT_DIR/libvirt/xml
     - scp -o StrictHostKeyChecking=no -i $AWS_EC2_SSH_KEY_FILE "ubuntu@$INSTANCE_IP:/tmp/qemu-ddvm-*.log" $CI_PROJECT_DIR/libvirt/qemu
     - scp -o StrictHostKeyChecking=no -i $AWS_EC2_SSH_KEY_FILE "ubuntu@$INSTANCE_IP:/tmp/dnsmasq/*" $CI_PROJECT_DIR/libvirt/dnsmasq
+    # build vm-metrics collector
+    - |
+      GO_ARCH=$ARCH
+      if [ "${ARCH}" == "x86_64" ]; then
+        GO_ARCH=amd64
+      fi
+    - cd test/new-e2e && GOOS=linux GOARCH="${GO_ARCH}" go build system-probe/vm-metrics/vm-metrics.go
+    # upload vm-metrics collector to metal instance
+    - scp -o StrictHostKeyChecking=no -i $AWS_EC2_SSH_KEY_FILE $CI_PROJECT_DIR/test/new-e2e/vm-metrics "ubuntu@$INSTANCE_IP:/home/ubuntu/vm-metrics"
+    # run vm-metrics collector
+    - ssh -o StrictHostKeyChecking=no -i $AWS_EC2_SSH_KEY_FILE "ubuntu@$INSTANCE_IP" "/home/ubuntu/vm-metrics -statsd-host=127.0.0.1 -statsd-port=8125 -libvirt-uri=/var/run/libvirt/libvirt-sock-ro --tag \"arch:${ARCH}\" --tag \"ci-pipeline-id:${CI_PIPELINE_ID}\" --daemon -log-file /home/ubuntu/daemon.log"
```

Comment:
Shouldn't this be in the `.upload_{sysprobe,secagent}_tests` jobs, as the vm-metrics program is already prepared as a dependency there?

Reply:
If we place it in `.upload_{sysprobe,secagent}_tests`, then running it becomes complicated.
To run the vm-metrics collector, the `datadog-agent` must be running which happens in the `setup_env` stage. So we can run only after this stage is complete. So either in `setup_env` or `run_tests`.
If we run in `setup_env` the `upload_tests` stage may not have been completed in time. If we pick `run_tests`, we need to have some extra logic to ensure that we do not run the collector multiple times.

So the easiest options seems to be to just upload and run the collector in `setup_env`

Reply:
Sounds good. I'd add a comment explaining why it needs to go there, just in case :D